src/mpl/src/gpu/mpl_gpu_ze.c

```diff
@@ -279,6 +283,7 @@ int MPL_gpu_finalize(void)
     MPL_free(local_to_global_map);
     MPL_free(global_to_local_map);
     MPL_free(global_ze_devices_handle);
+    MPL_free(shared_device_fds);
```

Comment:
If we move this variable inside `ipc_fd.c`, it can be cleaned up in `MPIDI_FD_mpi_finalize_hook`, right?

Reply:
`shared_device_fds` can't be moved into `ipc_fd.c` because it needs to be used inside `mpl_gpu_ze.c` during IPC API calls (see https://github.com/pmodels/mpich/pull/6424/commits/285881d9249577aad49703913e17ef884b3fbc6d).